src/components/settings-menu/NotificationsSection.tsx

```diff
@@ -1,122 +1,401 @@
-            />
-          }
-          onPress={() => onPress('pugson.eth')}
-          size={52}
-          titleComponent={<MenuItem.Title text="pugson.eth" />}
-        />
-      </Menu>
-      <Menu>
-        <MenuItem
-          disabled
-          rightComponent={<Switch />}
-          size={52}
-          titleComponent={
-            <MenuItem.Title
-              text={lang.t('settings.notifications_section.watched_wallets')}
-              weight="bold"
-            />
-          }
-        />
-        <MenuItem
-          hasRightArrow
-          labelComponent={
-            <MenuItem.Label
-              text={lang.t('settings.notifications_section.off')}
-            />
-          }
-          leftComponent={
-            <ContactAvatar
-              alignSelf="center"
-              color={colors.red}
-              marginRight={10}
-              size="small"
-              value="moxey.eth"
-            />
-          }
-          onPress={() => onPress('moxey.eth')}
-          size={52}
-          titleComponent={<MenuItem.Title text="moxey.eth" />}
-        />
-      </Menu>
-    </MenuContainer>
+          )}
+        </Box>
+      }
+      onPress={() => navigateToWalletSettings(walletName, wallet.address)}
+      size={52}
+      titleComponent={<MenuItem.Title text={walletName} />}
+    />
+  );
+};
+
+const NotificationsSection = () => {
+  const { navigate } = useNavigation();
+  const { colors } = useTheme();
+  const { chainId } = useAccountSettings();
+  const notMainnet = chainId !== 1;
+  const { wallets } = useWallets();
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const walletIDs = Object.keys(wallets!);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const ownedWallets: any[] = useMemo(() => [], []);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const watchedWallets: any[] = useMemo(() => [], []);
+  const [walletsLoaded, setWalletsLoaded] = useState(false);
+  const {
+    ownerEnabled,
+    watcherEnabled,
+    updateGroupSettings,
+  } = useWalletGroupNotificationSettings();
+  const noOwnedWallets = walletsLoaded && !ownedWallets.length;
+  const noWatchedWallets = walletsLoaded && !watchedWallets.length;
+  const [permissionStatus, setPermissionStatus] = useState<string | boolean>(
+    false
+  );
+  const noPermissions = permissionStatus !== RESULTS.GRANTED;
+
+  useEffect(() => {
+    // group wallets by relationship if the arrays are empty on first load
+    if (watchedWallets.length < 1 && ownedWallets.length < 1) {
+      walletIDs.forEach(key => {
+        const wallet = wallets?.[key];
+
+        if (wallet?.type === WalletTypes.readOnly) {
+          wallet?.addresses.forEach(item => watchedWallets.push(item));
+        } else {
+          wallet?.addresses.forEach(item => ownedWallets.push(item));
+        }
+      });
+    }
+
+    setWalletsLoaded(true);
+
+    checkNotifications().then(({ status }) => {
```

Comment:
this part actually can stay in `useEffect`, cause this is not causing any triggers in state